tsuru/resource_tsuru_app.go

```diff
@@ -0,0 +1,370 @@
+	if err := validPlan(ctx, provider, plan); err != nil {
+		return diag.FromErr(err)
+	}
+
+	tags := []string{}
+	for _, item := range d.Get("tags").([]interface{}) {
+		tags = append(tags, item.(string))
+	}
+
+	app := tsuru_client.InputApp{
+		Name:      d.Get("name").(string),
+		Platform:  platform,
+		Pool:      pool,
+		Plan:      plan,
+		TeamOwner: d.Get("team_owner").(string),
+		Tags:      tags,
+	}
+
+	if m, ok := d.GetOk("metadata"); ok {
+		metadata := metadataFromResourceData(m)
+		if metadata != nil {
+			app.Metadata = *metadata
+		}
+	}
+
+	if desc, ok := d.GetOk("description"); ok {
+		app.Description = desc.(string)
+	}
+
+	_, _, err := provider.TsuruClient.AppApi.AppCreate(ctx, app)
+	if err != nil {
+		return diag.Errorf("unable to create app %s: %v", app.Name, err)
+	}
+
+	d.SetId(app.Name)
+
+	return resourceTsuruApplicationRead(ctx, d, meta)
+}
+
+func resourceTsuruApplicationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	provider := meta.(*tsuruProvider)
+	name := d.Get("name").(string)
+	platform := d.Get("platform").(string)
+	if err := validPlatform(ctx, provider, platform); err != nil {
+		return diag.FromErr(err)
+	}
+
+	pool := d.Get("pool").(string)
+	if err := validPool(ctx, provider, pool); err != nil {
+		return diag.FromErr(err)
+	}
+
+	plan := d.Get("plan").(string)
+	if err := validPlan(ctx, provider, plan); err != nil {
+		return diag.FromErr(err)
+	}
+
+	tags := []string{}
+	for _, item := range d.Get("tags").([]interface{}) {
+		tags = append(tags, item.(string))
+	}
+
+	app := tsuru_client.UpdateApp{
+		Platform:  platform,
+		Pool:      pool,
+		Plan:      plan,
+		TeamOwner: d.Get("team_owner").(string),
+		Tags:      tags,
+	}
+
+	if m, ok := d.GetOk("metadata"); ok {
+		metadata := metadataFromResourceData(m)
+		if metadata != nil {
+			app.Metadata = *metadata
+		}
+	}
+
+	if desc, ok := d.GetOk("description"); ok {
+		app.Description = desc.(string)
+	}
+
+	restart := true
+	if r, ok := d.GetOk("restart_on_update"); ok {
+		restart = r.(bool)
+	}
+
+	if !restart {
+		app.NoRestart = true
+	}
+
+	_, err := provider.TsuruClient.AppApi.AppUpdate(ctx, name, app)
+	if err != nil {
+		return diag.Errorf("unable to update app %s: %v", name, err)
+	}
+
+	return nil
+}
+
+func resourceTsuruApplicationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	provider := meta.(*tsuruProvider)
+	name := d.Id()
+
+	app, _, err := provider.TsuruClient.AppApi.AppGet(ctx, name)
+	if err != nil {
+		return diag.Errorf("unable to create app %s: %v", app.Name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("platform", app.Platform)
+	d.Set("pool", app.Pool)
+	d.Set("plan", app.Plan.Name)
+	d.Set("team_owner", app.TeamOwner)
+
+	if app.Description != "" {
+		d.Set("description", app.Description)
+	}
+
+	if len(app.Tags) > 0 {
+		tags := []interface{}{}
+		for _, item := range app.Tags {
+			tags = append(tags, item)
+		}
+		d.Set("tags", tags)
+	}
```

Comment:
@pedrokiefer we dont need a conversion
```suggestion
	d.Set("tags", app.Tags)
```